information/specification.md

```diff
@@ -0,0 +1,706 @@
+algorithm to have. These properties include:
+
+ * A coin selection algorithm should, over the course of time, aim to generate
+   and maintain a UTxO set with _useful_ outputs: that is, outputs that allow
+   us to process future payments with a _minimum_ number of inputs.
+
+ * A coin selection algorithm should employ strategies to limit the
+   amount of _dust_ that accumulates in the UTxO set.
+
+For more information on dust avoidance, see [Self Organisation in Coin
+Selection](#self-organization-in-coin-selection).
+
+# Definitions
+
+### Address
+
+An _address_ is a unique identifier that represents a destination for a payment.
+
+Addresses are owned (and generated) by individual wallets.
+
+In general, coin selection algorithms are agnostic to the choice of address
+format. Therefore, the details of individual address formats are not described
+within this document.
+
+### Coin Amount
+
+A _coin amount_ is a non-negative integer value that represents a number of
+[Lovelace](https://cardanodocs.com/cardano/monetary-policy/).
+
+One [Ada](https://cardanodocs.com/cardano/monetary-policy/) is _exactly_ equal
+to one million Lovelace.
+
+### UTxO Set
+
+A _UTxO set_ represents the unspent value associated with a wallet.
+
+Each member of the set is a pair of the form (**_i_**, **_o_**), where:
+
+  * **_i_** is a [transaction input](#transaction-input);
+  * **_o_** is a [transaction output](#transaction-output).
+
+A given [transaction input](#transaction-input) can only appear _once_ in a
+given UTxO set.
+
+### Transaction Input
+
+A _transaction input_ is a pair of values (**_h_**, **_n_**), where:
+
+  * **_h_** is the hash of an existing transaction **_t_**;
+  * **_n_** is a 0-based integer index into the output list of transaction
+    **_t_**.
+
+### Transaction Output
+
+A _transaction output_ is a pair of values (**_t_**, **_a_**), where:
+
+  * **_t_** is a [target address](#address);
+  * **_a_** is a [coin amount](#coin-amount).
+
+# Interface
+
+All coin selection algorithms used by Cardano Wallet implement a _common
+interface_.
+
+Essentially, a coin selection algorithm is simply a _mathematical function_
+that, when applied to a [standard set of parameters](#parameters), will produce
+a [standard set of results](#results).
+
+This section defines the [parameters](#parameters) accepted by coin selection
+algorithms used in Cardano Wallet, the [results](#results) returned, as well as
+the various [error conditions](#failure-modes) that may occur.
+
+In this section, the terms _coin selection algorithm_ and _coin selection
+function_ will be used interchangeably.
+
+## Parameters
+
+A coin selection function accepts the following parameters:
+
+ 1. #### Requested Output List
+
+    A list of payments to be made to recipient addresses, encoded as a list of
+    [transaction outputs](#transaction-output).
+
+ 2. #### Initial UTxO Set
+
+    A subset of a wallet's [UTxO set](#utxo-set).
```

Comment:
You could say "The wallet's UTxO set, or a subset of it." Then the last sentence won't be necessary.

Reply:
Or "A UTxO set to choose from", that it comes from a wallet is quite anecdotal.

Reply:
@rvl @KtorZ 

I've attempted to address this issue here: 4ea2f9089c135796f8fdd873a2749eff1c9b3516

Hopefully the new wording is sufficiently general, but let me know if you think it could be improved further. :+1: